src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala

```diff
@@ -59,115 +73,191 @@ trait RawlsBillingProjectComponent {
+      val ownerSubgroupsQuery = findProjectSubgroupsWithRole(rawlsProjectName, ProjectRoles.Owner)
+      val userSubgroupsQuery = findProjectSubgroupsWithRole(rawlsProjectName, ProjectRoles.User)
+
+      val owners = ownerUsersQuery union ownerSubgroupsQuery
+      val users = userUsersQuery union userSubgroupsQuery
 
-      query.result.map(_.map { case (email, role) =>
-        RawlsBillingProjectMember(RawlsUserEmail(email), ProjectRoles.withName(role))
-      })
+      (owners union users).result.map(_.map{ case(email, role) => RawlsBillingProjectMember(RawlsUserEmail(email), ProjectRoles.withName(role))})
     }
 
     def delete(billingProject: RawlsBillingProject): ReadWriteAction[Boolean] = {
-      val name = billingProject.projectName.value
-      val projectQuery = findBillingProjectByName(name)
-
-      uniqueResult[RawlsBillingProjectRecord](projectQuery).flatMap {
-        case None => DBIO.successful(false)
-        case Some(projectRec) =>
-          findUsersByProjectName(name).delete andThen projectQuery.delete map { count => count > 0 }
-      }
+      rawlsBillingProjectGroupQuery.filter(_.projectName === billingProject.projectName.value).delete andThen
+        rawlsGroupQuery.delete(billingProject.owners) andThen rawlsGroupQuery.delete(billingProject.users) andThen
+        rawlsBillingProjectQuery.filter(_.projectName === billingProject.projectName.value).delete map { count => count > 0 }
     }
 
-    def addUserToProject(userRef: RawlsUserRef, billingProjectName: RawlsBillingProjectName, role: ProjectRole): WriteAction[ProjectUsersRecord] = {
-      val record = marshalProjectUsers(userRef, billingProjectName, role)
-      projectUsersQuery insertOrUpdate record map { _ => record }
+    //this should probably end up unused too
+    def addUserToProject(member: Either[RawlsUserRef, RawlsGroupRef], billingProjectName: RawlsBillingProjectName, role: ProjectRole): ReadWriteAction[RawlsGroup] = {
+      rawlsGroupQuery.load(RawlsGroupRef(RawlsGroupName(toBillingProjectGroupName(billingProjectName, role)))) flatMap {
+        case None => throw new RawlsException(s"Unable to save member ${member} as ${role} on project ${billingProjectName}")
+        case Some(group) => member match {
+          case Left(userToAdd) => rawlsGroupQuery.save(group.copy(users = group.users ++ Set(userToAdd)))
+          case Right(groupToAdd) => rawlsGroupQuery.save(group.copy(subGroups = group.subGroups ++ Set(groupToAdd)))
+        }
+      }
     }
 
-    def removeUserFromProject(userRef: RawlsUserRef, billingProject: RawlsBillingProject): WriteAction[Boolean] = {
-      val query = projectUsersQuery.filter(q => q.userSubjectId === userRef.userSubjectId.value && q.projectName === billingProject.projectName.value)
-      query.delete.map { count => count > 0 }
+//    def removeUserFromProject(member: Either[RawlsUserRef, RawlsGroupRef], billingProjectName: RawlsBillingProjectName): ReadWriteAction[Boolean] = {
```

Comment:
add/remove users from projects is now handled by updateGroupMembers in UserService